convention-plugins/src/main/kotlin/com.handstandsam.jvm.lib.gradle.kts

```diff
@@ -1,16 +1,42 @@
+import org.jetbrains.kotlin.gradle.plugin.KotlinSourceSet
+
 plugins {
     kotlin("multiplatform")
 }
 
 kotlin {
+    jvm()
     sourceSets {
-        val main by creating {
-            kotlin.srcDir("src/main/java")
-        }
         val commonMain by getting {
-            dependsOn(main)
+            kotlin.srcDirs("src/main/java", "src/main/kotlin")
+        }
+
+        val commonTest by getting {
+            kotlin.srcDirs("src/test/java", "src/test/kotlin")
+        }
+
+        val implementation by configurations.creating
+        val testImplementation by configurations.creating
+
+        configurations.getByName(commonMain.implementationConfigurationName).extendsFrom(implementation)
+        configurations.getByName(commonTest.implementationConfigurationName).extendsFrom(testImplementation)
+
+        fun createAliasConfigurations(sourceSet: KotlinSourceSet) {
+            sourceSet.relatedConfigurationNames.forEach { configurationName ->
+                val aliasName = configurationName
+                    .removePrefix("common")
+                    .removePrefix("Main")
+                    .decapitalize()
+
+                val configuration = configurations.getByName(configurationName)
+                val aliasConfiguration = configurations.maybeCreate(aliasName)
+                configuration.extendsFrom(aliasConfiguration)
+            }
```

Comment:
This is a bit "hacky".  There are a limited number of supported configurations, and they should be defined explicitly like the ones above instead of using this method.